Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrExtensions.kt

```diff
@@ -70,4 +70,20 @@ val ClassStub.explicitPrimaryConstructor: ConstructorStub?
     get() = functions.filterIsInstance<ConstructorStub>().firstOrNull(ConstructorStub::isPrimary)
 
 fun ClassStub.nestedName(): String =
-        classifier.relativeFqName.substringAfterLast('.')
\ No newline at end of file
+        classifier.relativeFqName.substringAfterLast('.')
+
+fun ConstantStub.determineEnumEntryClassifier(): Classifier = when (this) {
+    is StringConstantStub -> "String"
+    is IntegralConstantStub -> when (size) {
+        1 -> if (isSigned) "Byte" else "UByte"
+        2 -> if (isSigned) "Short" else "UShort"
+        4 -> if (isSigned) "Int" else "UInt"
+        8 -> if (isSigned) "Long" else "ULong"
+        else -> error("Integral constant with unexpected size of $size.")
+    }
+    is DoubleConstantStub -> when (size) {
```

Comment:
Could enum entry be of non-integral type?

Reply:
No. This method will be reused for other constants (e.g. macros). I'll rename it.